GVFS/GVFS/RepairJobs/GitRefsRepairJob.cs

```diff
@@ -0,0 +1,163 @@
+﻿using GVFS.Common;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.RepairJobs
+{
+    public abstract class GitRefsRepairJob : RepairJob
+    {
+        protected GitRefsRepairJob(ITracer tracer, TextWriter output, GVFSEnlistment enlistment) 
+            : base(tracer, output, enlistment)
+        {
+        }
+
+        public override IssueType HasIssue(List<string> messages)
+        {
+            int numBadRefs = 0;
+
+            foreach (string @ref in this.GetRefs())
+            {
+                if (!this.TryParseRef(@ref, messages))
+                {
+                    numBadRefs++;
+                }
+            }
+
+            return numBadRefs > 0 ? IssueType.Fixable : IssueType.None;
+        }
+
+        /// <summary>
+        /// Fixes the refs using the reflog to find the last SHA.
+        /// </summary>
+        public override FixResult TryFixIssues(List<string> messages)
+        {
+            IEnumerable<string> badRefs = this.GetRefs().Where(x => !this.TryParseRef(x));

```

Comment:
Why did you use a `foreach` above and then use Linq here?  Can we be use the same in both?